src/examples/IK/CMakeLists.txt

```diff
@@ -1,4 +1,7 @@
 
+include_directories(${CMAKE_CURRENT_BINARY_DIR})
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/ConfigFolderPath.h.in" "${CMAKE_CURRENT_BINARY_DIR}/ConfigFolderPath.h" @ONLY)
+
 find_package(matioCpp REQUIRED)
 
 add_executable(exampleIK)
```

Comment:
Not something related to this PR or something that is blocking this PR to be merged, but typically it is a good idea to have examples separated from the main CMake build of the project, see for example https://github.com/robotology/idyntree/blob/master/examples/cxx/InverseKinematics/CMakeLists.txt . The idea is that if the example are self-contained, the users can just copy them and start working on them. Otherwise, it may not be obvious for users (especially not expert in CMake) how they should change the CMake of the example to run on its onw.

Reply:
I tried to do it but when I added `find_package(BiomechanicalAnalysisFramework REQUIRED)` to the CMakeLists.txt I got the following error:

```
CMake Error at build/BiomechanicalAnalysisFrameworkConfig.cmake:22 (get_property):
  get_property could not find TARGET IBPFNAMESPACE{_target}.  Perhaps it has
  not yet been created.
Call Stack (most recent call first):
  src/examples/IK/CMakeLists.txt:13 (find_package)
```

and I noticed that the cmake macro [AddBiomechanicalAnalysisFrameworkTarget.cmake](https://github.com/ami-iit/biomechanical-analysis-framework/blob/main/cmake/AddBiomechanicalAnalysisFrameworkTarget.cmake) is different from the one in `blf` (from which I supposed it was copied). I tried to copy and paste the one in blf ([this one]((https://github.com/ami-iit/bipedal-locomotion-framework/blob/master/cmake/AddBipedalLocomotionFrameworkTarget.cmake)) and it solved the problem. Moreover, to make it work I also had to copy and paste the macro [InstallBasicPackageFiles.cmake](https://github.com/ami-iit/bipedal-locomotion-framework/blob/master/cmake/InstallBasicPackageFiles.cmake) from `blf`.

Reply:
However, I would merge this PR and then open a new one to update the cmake functions.

Reply:
I opened #23 to do it.